h/services/oauth.py

```diff
@@ -161,10 +144,51 @@ def create_token(self, user, authclient):
 
         return token
 
-    def _decode(self, token, **kwargs):
+
+class GrantToken(object):
+    """
+    Represents a JWT bearer grant token.
+
+    This class is responsible for a couple of things: firstly, verifying that
+    the token is a correctly-formatted JSON Web Token, and that it contains all
+    the required claims in the right formats. Some of this processing is
+    deferred to the `jwt` module, but that doesn't handle all the fields we
+    want to validate.
+
+    """
+
+    def __init__(self, token):
+        if not token or type(token) != text_type:
+            raise OAuthTokenError('required assertion parameter is missing',
```

Comment:
I found it a bit weird to have this error message here, because the implementation of this class doesn't know about an assertion parameter, it just gets the token string. But it's probably good enough for now as long as we don't use the `GrantToken` class anywhere else than here.